psci/PSCi.hs

```diff
@@ -531,55 +343,7 @@ handleKindOf typ = do
             Right (kind, _) -> PSCI . outputStrLn . P.prettyPrintKind $ kind
         Nothing -> PSCI $ outputStrLn "Could not find kind"
 
--- Commands
-
--- |
--- Parses the input and returns either a Metacommand, or an error as a string.
---
-getCommand :: Bool -> InputT (StateT PSCiState IO) (Either String (Maybe Command))
-getCommand singleLineMode = handleInterrupt (return (Right Nothing)) $ do
-  firstLine <- withInterrupt $ getInputLine "> "
-  case firstLine of
-    Nothing -> return (Right (Just QuitPSCi)) -- Ctrl-D when input is empty
-    Just "" -> return (Right Nothing)
-    Just s | singleLineMode || head s == ':' -> return .fmap Just $ parseCommand s
-    Just s -> fmap Just . parseCommand <$> go [s]
-  where
-    go :: [String] -> InputT (StateT PSCiState IO) String
-    go ls = maybe (return . unlines $ reverse ls) (go . (:ls)) =<< getInputLine "  "
-
--- |
--- Performs an action for each meta-command given, and also for expressions.
---
-handleCommand :: Command -> PSCI ()
-handleCommand (Expression val) = handleExpression val
-handleCommand ShowHelp = PSCI $ outputStrLn helpMessage
-handleCommand (Import im) = handleImport im
-handleCommand (Decls l) = handleDecls l
-handleCommand (LoadFile filePath) = whenFileExists filePath $ \absPath -> do
-  m <- psciIO $ loadModule absPath
-  case m of
-    Left err -> PSCI $ outputStrLn err
-    Right mods -> PSCI . lift $ modify (updateModules (map (absPath,) mods))
-handleCommand (LoadForeign filePath) = whenFileExists filePath $ \absPath -> do
-  foreignsOrError <- psciIO . runMake $ do
-    foreignFile <- makeIO (const (P.ErrorMessage [] $ P.CannotReadFile absPath)) (readFile absPath)
-    P.parseForeignModulesFromFiles [(absPath, foreignFile)]
-  case foreignsOrError of
-    Left err -> PSCI $ outputStrLn $ P.prettyPrintMultipleErrors False err
-    Right foreigns -> PSCI . lift $ modify (updateForeignFiles foreigns)
-handleCommand ResetState = do
-  PSCI . lift . modify $ \st ->
-    st { psciImportedModules = []
-       , psciLetBindings     = []
-       }
-  loadAllImportedModules
-handleCommand (TypeOf val) = handleTypeOf val
-handleCommand (KindOf typ) = handleKindOf typ
-handleCommand (BrowseModule moduleName) = handleBrowse moduleName
-handleCommand (ShowInfo QueryLoaded) = handleShowLoadedModules
-handleCommand (ShowInfo QueryImport) = handleShowImportedModules
-handleCommand QuitPSCi = P.internalError "`handleCommand QuitPSCi` was called. This is a bug."
+-- Misc
 
 whenFileExists :: FilePath -> (FilePath -> PSCI ()) -> PSCI ()
```

Comment:
Ah, `PSCI` has no `MonadIO` instance. @paf31 How do you feel about using a derived `MonadIO` instance for `PSCI`? I'm pretty sure it would end up being the same as `psciIO`.